internal/serverinstall/docker.go

```diff
@@ -645,13 +646,17 @@ func (i *DockerInstaller) InstallRunner(
 		AttachStdin:  true,
 		OpenStdin:    true,
 		StdinOnce:    true,
+		User:         "root",
 		Image:        i.config.serverImage,
 		Env:          opts.AdvertiseClient.Env(),
-		Cmd:          []string{"runner", "agent", "-vvv"},
+		Cmd:          []string{"runner", "agent", "-vv"},
 		Labels: map[string]string{
 			"waypoint-type": "runner",
 		},
 	}, &container.HostConfig{
+		Privileged: true,
+		CapAdd:     []string{"CAP_DAC_OVERRIDE"},
+		Binds:      []string{"/var/run/docker.sock:/var/run/docker.sock"},
```

Comment:
I feel like introducing on-demand runners makes this less necessary. After 0.6, I feel like a normal installation should have a static runner that launches on-demand runners in a platform like k8s or ecs.

I think a side-effect of this change is that there will be a lot more mystery orphaned containers on their k8s workers from people accidentally using the docker plugin with remote runners. It will also make it harder for some security-focused companies to adopt waypoint.

Reply:
Not true because without this change, the static docker runner can't launch tasks, and thusly can't launch ondemand runners.

Reply:
Oh gotcha, if waypoint is installed on docker, then on-demand runners won't work there. Makes sense.

We could update this later to only add these capabilities if we're installed on docker. Maybe add `runner-caps` and `runner-binds` to the server install commands or something, which we set automatically when installing on docker..